### PR TITLE
Feature/edit graph

### PR DESCRIPTION
**Pull request by YahelR**

### CODE REVIEW

File: src/components/GraphWorkshop/GraphWorkshop.tsx [hunk 1]
```diff
@@ -1,25 +1,26 @@
 import { Modal } from "@mui/material";
-import ParametersPanel from "./ParametersPanel/ParametersPanel";
-import './GraphWorkshop.scss';
 import { Dispatch, SetStateAction } from "react";
+import { Graph } from "../../types/graph.types";
+import './GraphWorkshop.scss';
+import ParametersPanel from "./ParametersPanel/ParametersPanel";
```
Comment (netashefer) [inline src/components/GraphWorkshop/GraphWorkshop.tsx:4]: order

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx [hunk 2]
```diff
@@ -15,22 +15,45 @@ import MultipleSelect from "../../MultiSelect/MultiSelect";
...
-	const [dataFields, setDataFields] = useState<string[]>(null);
+	const [dataFields, setDataFields] = useState<string[]>(graphToEdit ? graphToEdit?.graphConfig?.dataFields : null);
 	const addGraphToDashboard = useAddWidget();
 
 	const addGraphToExistingGraphs = useRecoilCallback(({ set }) => (graph: Graph) => {
 		set(graphsAtom, prevState => [...prevState, graph]);
 	}, []);
 
+	const editGraphInExistingGraphs = useRecoilCallback(({set}) => (graph: Graph) =>{
+		set(graphsAtom, prevState => [...prevState.filter(g => g.graphId !== graph.graphId), graph])
```
Comment (netashefer) [inline src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: ;

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx [hunk 3]
```diff
@@ -59,18 +82,36 @@ const ParametersPanel = () => {
...
 
-	const chartConfig = chartMapping[graphType?.value as GraphType];
+	const editGraph = async (graph: Graph) => {
+		try {
+			const graphWithGraphId = {...graph, graphId: graphToEdit.graphId};
+			await graphCommunicator.editGraph(graphWithGraphId);
+			editGraphInExistingGraphs(graphWithGraphId);
+			notifySuccess("Sucessfully saved your edited graph!");
+		} catch (e) {
+			console.log(e);
```
Comment (netashefer) [inline src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: consolr.error

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx [hunk 4]
```diff
@@ -107,7 +148,7 @@ const ParametersPanel = () => {
 					<RenderIf condition={yAxis?.value === Aggragation.uniqueValues}>
 						<Dropdown
 							value={yAxisField}
-							onChange={setYAxisField}
+							onChange={(yAxisField) => {setYAxisField(yAxisField); setDataFields(null)}}
```
Comment (netashefer) [inline src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: create common function that get a callback (setYAxisField) and does setDataFields(null) by itself.
use it in those 3 places

File: src/components/WidgetContainer/WidgetContainer.tsx [hunk 5]
```diff
@@ -8,19 +8,22 @@ import { useRemoveWidget } from '../../recoil/customHooks/useWidgetHandler';
...
 
 const WidgetContainer = ({ widgetId, graphId, widgetProps }: WidgetContainerProps) => {
+	const [shouldOpenGraphWorkshop, setShouldOpenGraphWorkshop] = useState(false);
     const removeWidget = useRemoveWidget();
     const containerRef = useRef();
     const [width, setWidth] = useState(0);
     const [height, setHeight] = useState(0);
     const graph = useRecoilValue(graphSelector(graphId));
 
+	console.log(graph)
```
Comment (netashefer) [inline src/components/WidgetContainer/WidgetContainer.tsx]: --

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx (hunk 4 above)
Comment (YahelR) [reply src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: wouldn't it be a very specific function? how should I call it

Comment (netashefer) [reply src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: overrideFieldBy(callback)  maybe?

**Pull request merged**